### ISSUE TITLE
Local video is showing  on Remote view  for few seconds after room

### ISSUE THREAD

**Issue by MukFlex** [twilio/video-quickstart-android#406]
> After accepting the call, local video is showing on both remote(primaryVideoView) and local (thumbnailVideoView) views for a few seconds.

>Please suggest how to avoid local video on remote(primaryVideoView) after room connection.

 
>    private void moveLocalVideoToThumbnailView() {
      if (thumbnailVideoView.getVisibility() == View.GONE) {
                thumbnailVideoView.setVisibility(View.VISIBLE);
                localVideoTrack.removeRenderer(primaryVideoView);
                localVideoTrack.addRenderer(thumbnailVideoView);
                localVideoView = thumbnailVideoView;
                thumbnailVideoView.setMirror(cameraCapturerCompat.getCameraSource() ==
                        CameraCapturer.CameraSource.FRONT_CAMERA);
            }
         }

**Comment by Gryzor** [comment]
> Please suggest how to avoid local video on remote(primaryVideoView) after room connection.
I don't understand if your problem is that you **do not want** your local video to show on the remote feed or if it's not showing and you want it to show. 

For the former, if you *don't want* to send your video feed, you don't have to...

Look at the sample: https://github.com/twilio/video-quickstart-android/blob/master/quickstartKotlin/src/main/java/com/twilio/video/quickstart/kotlin/VideoActivity.kt

```
 /*
         * If connected to a Room then share the local video track.
         */
        localVideoTrack?.let { localParticipant?.publishTrack(it) }
```

^  don't share your video track if you don't want to.
But you can also turn your local feed on/off at will. Look at the same sample:
```
private fun localVideoClickListener(): View.OnClickListener
```

Inside you can see that it does: (code simplified for brevity)
```
localVideoTrack?.let {
                val enable = !it.isEnabled
                it.enable(enable)
}
```
in other words: `localVideoTrack?.enable(false)` 

Is that what you mean?

**Comment by zackm0571** [comment]
Hey @MukFlex, 
One approach you could take is to wait until a frame is available from a `RemoteParticipant` before moving the `LocalParticipant` video to the primary view.  You can do this by attaching a `VideoRenderer.Listener` to a `RemoteParticipant` thumbnail view if the room contains more than one participant. 

```
            thumbnailVideoView.setListener(new VideoRenderer.Listener() {
                @Override
                public void onFirstFrame() {

                    // Attach track to primary view
                    // or toggle flag that indicates
                    // the frame is ready to be rendered
                }

                @Override
                public void onFrameDimensionsChanged(int width, int height, int rotation) {

                }
            });
        }
```

What do you think of this, @MukFlex?

**Comment by MukFlex** [comment]
@zackm0571 Thanks. I will try it.

**Comment by MukFlex** [comment]
> Hey @MukFlex,
> One approach you could take is to wait until a frame is available from a `RemoteParticipant` before moving the `LocalParticipant` video to the primary view. You can do this by attaching a `VideoRenderer.Listener` to a `RemoteParticipant` thumbnail view if the room contains more than one participant.
> 
> ```
>             thumbnailVideoView.setListener(new VideoRenderer.Listener() {
>                 @Override
>                 public void onFirstFrame() {
> 
>                     // Attach track to primary view
>                     // or toggle flag that indicates
>                     // the frame is ready to be rendered
>                 }
> 
>                 @Override
>                 public void onFrameDimensionsChanged(int width, int height, int rotation) {
> 
>                 }
>             });
>         }
> ```
> 
> What do you think of this, @MukFlex?

I  am tried to show black screen on the primary view until receiving any RemoteParticipant video frame on other side.

**Comment by zackm0571** [comment]
Hey @MukFlex, 

Where are you at this with one? Any luck getting it to function as expected?

**Comment by aaalaniz** [comment]
I'm closing this issue for now. Please reopen if you are continuing to experience problems delaying the rendering of video.

Thank you
